.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -251,29 +288,37 @@ function generateReport({ issuesData, previousReport, componentData }) {
     const section = [``];
     const data = byStatus[lbl];
     const count = Object.keys(data).length;
-
-    section.push(`<li> ${lbl}: ${count}`);
+    let li = `<li> ${lbl}: ${count}`; 
+    if (previousReport !== null) {
+      li = addChangesFromPreviousWeek(li, count, previousReport.componentData[lbl])
+    }
+    section.push(li)
     if (data.length !== 0) {
     // NOTE: the newline after <summary> is required for markdown to render correctly
       section.push(`<details>
     <summary> Components </summary>\n`);
       section.push(`${Object.keys(data).map((compName) => {
         const {stability} = data[compName]
         return `- [ ] ${compName}: ${JSON.stringify(stability)}`
-      }).join('\n')}
-</details>`);
+      }).join('\n')}`)
+      section.push(`</details>`);
     }
     section.push('</li>');
     out.push(section.join('\n'));
   }
 
-  let {count, section} = generateComponentsLookingForOwnersReportSection(lookingForOwners)
-  out.push(`<li> Seeking new code owners: ${count}`)
+  let {lookingForOwnersCount, section} = generateComponentsLookingForOwnersReportSection(lookingForOwners)
+  li = `<li> Seeking new code owners: ${lookingForOwnersCount}`
+  if (previousReport !== null) {
+    li = addChangesFromPreviousWeek(li, lookingForOwnersCount, previousReport.componentData.lookingForOwners)
+  }
+  out.push(li)
   out.push(...section)
   out.push('</li>')
   out.push('</ul>');
 
-  const report = out.join('\n');
+  let report = out.join('\n');
+  report = tryAddJSONData(report, issuesData, byStatus, lookingForOwnersCount)
```

Comment:
Since the length check is vital to this action passing, should we add a comment here to try to make it clear that the `tryAddJSONData` check needs to be last? My concern is adding more data after could cause the report to be too long, causing the action to fail.

Maybe there's a better way than a comment, but that was my first thought.

Reply:
How about now?
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38884/files#diff-6d378c7867e0f7f3eb58ea282fa1d63937c870c9e55825a4600d3f935592cbb5R322-R325